specutils/analysis/template_match.py

```diff
@@ -0,0 +1,36 @@
+from ..spectra.spectrum1d import Spectrum1D
+from ..spectra.spectrum_collection import SpectrumCollection
+from ..manipulation import FluxConservingResample
+from scipy.stats import chisquare
+
+def _template_match(observed_spectrum, template_spectrum):
+    # resample the template spectrum to match the wavelength of the observed spectrum
+    fluxc_resample = FluxConservingResample()
+    output_spectrum1D = fluxc_resample(template_spectrum, observed_spectrum.wavelength)
+
+    # calculate chi square
+    x2 = chisquare(observed_spectrum.flux, output_spectrum1D.flux)
+    return x2
+
+def template_match(observed_spectrum, collection):
```

Comment:
Sorry, when I wrote on the board I think I used `collection`. We might want to stay away from that name as `collections` (plural) is a python thing and people, like me, might get confused.  Maybe `templates` or `spectral_templates`?